src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala

```diff
@@ -490,51 +530,22 @@ object DeploymentValidation extends SparkSessionWrapper {
    * @param ehName
    */
   private def validateEventHub(
-                                workspace_id: String,
-                                scope: String,
-                                optKey: Option[String],
-                                optEHName: Option[String],
-                                outputPath: String
+                               config: MultiWorkspaceConfig
                               ): DeploymentValidationReport = {
-    if (optKey.isEmpty || optEHName.isEmpty) throw new BadConfigException("When cloud is Azure, the eh_name and " +
-      "eh_scope_key are required fields but they were empty in the config")
+
+    val isAAD = config.aad_client_id.nonEmpty && config.aad_tenant_id.nonEmpty && config.aad_client_secret_key.nonEmpty
```

Comment:
I think you're missing validation of the composite sets.

You need to validate that the config provides one of the two following states and derive isAAD from there -- you should create a stand-alone function for this and provide sufficient unit tests for positive / negative outcomes. If user violates these required conditions -- throw a user friendly error.

isEHAAD:
--- nonEmpty ---
aad_client_id
aad_tenant_id
aad_client_secret_key
eh_name
--- empty ---
eh_scope_key

!isEHAAD
--- nonEmpty ---
eh_scope_key
eh_name
--- empty --- 
aad_client_id
aad_tenant_id
aad_client_secret_key

Reply:
done